pkg/controllerhelpers/scylla.go

```diff
@@ -15,46 +16,78 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/util/errors"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
 	corev1schedulinghelpers "k8s.io/component-helpers/scheduling/corev1"
 	"k8s.io/component-helpers/scheduling/corev1/nodeaffinity"
 )
 
-func GetScyllaIPFromService(svc *corev1.Service) (string, error) {
-	if svc.Spec.Type != corev1.ServiceTypeClusterIP {
-		return "", fmt.Errorf("service %s is of type %q instead of %q", naming.ObjRef(svc), svc.Spec.Type, corev1.ServiceTypeClusterIP)
+func GetScyllaHost(ctx context.Context, sc *scyllav1.ScyllaCluster, coreV1Client corev1client.CoreV1Interface, svc *corev1.Service) (string, error) {
+	// Assume API's default.
+	if sc.Spec.ExposeOptions == nil || sc.Spec.ExposeOptions.BroadcastOptions == nil {
+		if svc.Spec.ClusterIP == corev1.ClusterIPNone {
+			return "", fmt.Errorf("service %q does not have a ClusterIP", naming.ObjRef(svc))
+		}
+
+		return svc.Spec.ClusterIP, nil
```

Comment:
this branch is also in `GetScyllaBroadcastAddress` - could this just default the broadcast option object to scyllav1.BroadcastAddressTypeServiceClusterIP?